docs/main.md

```diff
@@ -1,3 +1,18 @@
+- Each run (3 phase commit) of the consensus protocol orders a batch (collection) of transactions.
+- Nodes maintain several ledgers, each for a distinct purpose, it has a pool ledger for node membership transactions like addition of new node, suspension of a node, change of IP/port or keys of a node, a ledger for identity transactions, etc 
+- Apart for the ledger, nodes maintain state (for each ledger) which is [Merkle Patricia Trie](https://github.com/ethereum/wiki/wiki/Patricia-Tree). It might maintain several other projections of the ledger. For more on storage, refer the [storage document](storage.md). 
+- Clients with appropriate permissions can send write requests (transactions) to the nodes but any client can send read requests to the nodes.
+- Communicate between nodes and clients and node-to-node happens on [CurveZMQ](http://curvezmq.org/). The codebase has an abstraction called `Stack` that manages communication. It has several variants which offer different features.
+- On receiving transactions nodes perform some basic validation and broadcast the request to other nodes. This is called request propagation, more details in RBFT paper. 
+  Once the nodes realise that enough nodes have got the request, they consider the request ready for processing.
+  The primary node initiates a new round of consensus through a 3 phase commit process at the end of which all nodes add the transaction to their ledger and corresponding storages. More details on 3 phase commit in RBFT paper.
+  Different kinds of requests update different ledgers and storage layers. A detailed explanation of request handling is present [here](request_handling.md)
+- During the life of the system, nodes might crash and restart, get disconnected/re-connected, new nodes might join the network. Thus nodes need a mechanism to get the transactions they have missed or never had (new node). They do this by a process called catchup.
+  Here the nodes communicate their state and learn other node's states and then if the nodes realise that they are behind, they run a protocol to get the missing transactions. More on this in the [catchup document](catchup.md)
+- The nodes can request various protocol-specific messages from other nodes by a `MESSAGE_REQUEST` message.
+- When the primary node crashes (or becomes non functional in any way), or it misbehaves by sending bad messages or it slows down then the follower nodes initiate a protocol to change the leader, this protocol is called view change. 
+  View change involves selecting a new leader, which is done in a round robin fashion and communication (and catchup if needed) of state so before the new leader starts, every node has the same state. 
```

Comment:
Please add three more paragraphs explicitly describing how our consensus protocol differs from RBFT one:
1) BLS signatures
2) timestamps (set by Primary)
3) batches of requests

Reply:
Done